syntax = "proto3";

package statistics.protocol;

message KillStatistic {
  uint64 weapon_id = 1;
  uint64 count = 2;
}

message EnemyKillStatistics {
  uint64 enemy_id = 1;
  repeated KillStatistic per_weapon = 2;
  uint64 infight = 3;
  uint64 suicide = 4;
}

message KillStatistics { repeated EnemyKillStatistics per_enemy = 1; }

message Statistics { KillStatistics kill = 1; }
